lib/takeaway.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'menu'
+require_relative 'textsender'
+
+class Takeaway 
+  attr_reader :menu, :basket
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = Hash.new(0)
+    @sum = 0
+    
+  end
+
+  def view_menu
+    @menu.display
+  end
+
+  def select_dish(dish, quantity)
+    @basket[dish] += quantity
+  end
+
+  def total
+    @sum = @basket.reduce(0) do |sum, (dish, quantity)| 
+    sum + (@menu.menu[dish] * quantity)
```

Comment:
It might be better to have a method on the Menu class that allows you to get the dish rather than exposing the menu attribute and allowing people to change the menu whenever they like.